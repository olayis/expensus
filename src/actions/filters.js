// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
  type: 'SET_TEXT_FILTER',
  text,
});

// SORT_BY_DATE_RECENT
export const sortByDateRecent = () => ({
  type: 'SORT_BY_DATE_RECENT',
});

// SORT_BY_DATE_OLDEST
export const sortByDateOldest = () => ({
  type: 'SORT_BY_DATE_OLDEST',
});

// SORT_BY_AMOUNT_HIGHEST
export const sortByAmountHighest = () => ({
  type: 'SORT_BY_AMOUNT_HIGHEST',
});

// SORT_BY_AMOUNT_LOWEST
export const sortByAmountLowest = () => ({
  type: 'SORT_BY_AMOUNT_LOWEST',
});

// SORT_BY_DESCRIPTION_A_TO_Z
export const sortByDescriptionAtoZ = () => ({
  type: 'SORT_BY_DESCRIPTION_A_TO_Z',
});

// SORT_BY_DESCRIPTION_Z_TO_A
export const sortByDescriptionZtoA = () => ({
  type: 'SORT_BY_DESCRIPTION_Z_TO_A',
});

// SORT_BY_CATEGORY
export const sortByCategory = () => ({
  type: 'SORT_BY_CATEGORY',
});

// SET_START_DATE
export const setStartDate = (startDate) => ({
  type: 'SET_START_DATE',
  startDate,
});

// SET_END_DATE
export const setEndDate = (endDate) => ({
  type: 'SET_END_DATE',
  endDate,
});
