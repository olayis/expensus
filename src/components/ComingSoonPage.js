import React from 'react';
import { Link } from 'react-router-dom';
import programming from '../../public/img/programming.svg';

const ComingSoonPage = () => (
  <main className='main-content'>
    <div className='content-container'>
      <div className='error'>
        <img
          src={programming}
          alt='Programmer'
          className='error__illustration'
          width='1075'
          height='585'
        />
        <h2 className='error__text'>This page is coming soon.</h2>
        <div className='error__explanation'>
          Continue managing your expenses.
          <span className='error__action'>
            <Link className='btn btn--try-again' to='/'>
              Go Home
            </Link>
          </span>
        </div>
      </div>
    </div>
  </main>
);

export default ComingSoonPage;
