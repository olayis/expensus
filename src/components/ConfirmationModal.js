import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => {
  if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

  return (
    <div>
      <Modal
        contentLabel='Are you sure?'
        isOpen={!!props.deleteSelected}
        onRequestClose={props.stopDeleteDataSelected}
        closeTimeoutMS={200}
        className='modal'
        aria={{
          labelledby: 'heading',
          describedby: 'full_description',
        }}
      >
        <h3 id='heading' className='modal__title'>
          Delete Data
        </h3>
        <p id='full_description' className='modal__body'>
          Are you sure you want to delete all of your data on Expensus?
        </p>
        <button className='modal__btn--delete' onClick={props.handleDeleteData}>
          Delete
        </button>
        <button
          className='modal__btn--cancel'
          onClick={props.stopDeleteDataSelected}
        >
          Cancel
        </button>
      </Modal>
    </div>
  );
};

export default ConfirmationModal;
