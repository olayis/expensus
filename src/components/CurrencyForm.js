import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';

export default class CurrencyForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      currency: props.currency,
      currencyError: '',
    };
  }

  onCurrencyChange = (e) => {
    const currency = e.target.value;
    this.setState(() => ({ currency }));
  };

  onSubmit = (e) => {
    e.preventDefault();

    if (!this.state.currency) {
      this.setState(() => ({
        currencyError: 'Please select a valid currency',
      }));
    } else {
      this.setState(() => ({ currencyError: '' }));
      this.props.onSubmit({
        currency: this.state.currency.trim(),
      });
    }
  };

  render() {
    return (
      <form className='input-field' onSubmit={this.onSubmit}>
        <div className='input-field__wrapper'>
          <select
            className='dropdown dropdown--currency'
            value={this.state.currency}
            onChange={this.onCurrencyChange}
            aria-label='Select Currency'
            name='currency'
          >
            <option value=''>-- Select Currency --</option>
            <option value='Baht'>Baht (฿)</option>
            <option value='Danish Krone'>Danish Krone (DKK)</option>
            <option value='Dollar'>Dollar ($)</option>
            <option value='Dong'>Dong (₫)</option>
            <option value='Euro'>Euro (€)</option>
            <option value='Forint'>Forint (Ft)</option>
            <option value='Hryvnia'>Hryvnia (₴)</option>
            <option value='Koruna'>Koruna (Kč)</option>
            <option value='Lira'>Lira (₺)</option>
            <option value='Lev'>Lev (лв)</option>
            <option value='Naira'>Naira (₦)</option>
            <option value='Norwegian krone'>Norwegian krone (kr)</option>
            <option value='Pound Sterling'>Pound Sterling (£)</option>
            <option value='Rand'>Rand (R)</option>
            <option value='Real'>Real (R$)</option>
            <option value='Ruble'>Ruble (₽)</option>
            <option value='Swiss Franc'>Swiss Franc (CHF)</option>
            <option value='Yen'>Yen (¥)</option>
            <option value='Yuan'>Yuan (¥)</option>
            <option value='Zloty'>Złoty (PLN)</option>
          </select>
          {this.state.currencyError && (
            <span className='input-field__error-msg'>
              <FontAwesomeIcon className='fa-icon' icon={faExclamationCircle} />
              {this.state.currencyError}
            </span>
          )}
        </div>
        <button className='btn btn-currency'>Change Currency</button>
      </form>
    );
  }
}
