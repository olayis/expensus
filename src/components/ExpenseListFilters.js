import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import {
  setTextFilter,
  sortByDateRecent,
  sortByDateOldest,
  sortByAmountHighest,
  sortByAmountLowest,
  sortByDescriptionAtoZ,
  sortByDescriptionZtoA,
  sortByCategory,
  setStartDate,
  setEndDate,
} from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  };

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };

  onSortChange = (e) => {
    switch (e.target.value) {
      case 'date(recent)':
        this.props.sortByDateRecent();
        break;

      case 'date(oldest)':
        this.props.sortByDateOldest();
        break;

      case 'amount(highest)':
        this.props.sortByAmountHighest();
        break;

      case 'amount(lowest)':
        this.props.sortByAmountLowest();
        break;

      case 'description(a_to_z)':
        this.props.sortByDescriptionAtoZ();
        break;

      case 'description(z_to_a)':
        this.props.sortByDescriptionZtoA();
        break;

      case 'category':
        this.props.sortByCategory();
        break;
    }
  };

  render() {
    return (
      <div className='content-container'>
        <div className='filter-container'>
          <div className='input-group'>
            <div className='input-group__item'>
              <input
                className='text-input text-input--search'
                type='text'
                placeholder='Search'
                value={this.props.filters.text}
                onChange={this.onTextChange}
              />
              <FontAwesomeIcon
                icon={faSearch}
                className='text-input--search__icon'
              />
            </div>
            <div className='input-group__item dropdown'>
              <label className='text-input__label'>Sort by: </label>
              <select
                className='dropdown--sort'
                value={this.props.filters.sortBy}
                onChange={this.onSortChange}
              >
                <option value='date(recent)'>Date (Recent)</option>
                <option value='date(oldest)'>Date (Oldest)</option>
                <option value='amount(highest)'>Amount (Highest)</option>
                <option value='amount(lowest)'>Amount (Lowest)</option>
                <option value='description(a_to_z)'>A to Z</option>
                <option value='description(z_to_a)'>Z to A</option>
                <option value='category'>Category</option>
              </select>
            </div>
            <div className='input-group__item'>
              <DateRangePicker
                startDate={this.props.filters.startDate}
                startDateId='unique_start_date_id'
                endDate={this.props.filters.endDate}
                endDateId='unique_end_date_id'
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters,
});

const mapDispatchToProps = (dispatch) => ({
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate)),
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDateRecent: () => dispatch(sortByDateRecent()),
  sortByDateOldest: () => dispatch(sortByDateOldest()),
  sortByAmountHighest: () => dispatch(sortByAmountHighest()),
  sortByAmountLowest: () => dispatch(sortByAmountLowest()),
  sortByDescriptionAtoZ: () => dispatch(sortByDescriptionAtoZ()),
  sortByDescriptionZtoA: () => dispatch(sortByDescriptionZtoA()),
  sortByCategory: () => dispatch(sortByCategory()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
