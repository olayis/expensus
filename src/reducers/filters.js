// Filters Reducer
import moment from 'moment';

const filtersReducerDefaultState = {
  text: '',
  sortBy: 'date(recent)',
  startDate: moment().startOf('month'),
  endDate: moment().endOf('month'),
};

export default (state = filtersReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_TEXT_FILTER':
      return {
        ...state,
        text: action.text,
      };

    case 'SORT_BY_DATE_RECENT':
      return {
        ...state,
        sortBy: 'date(recent)',
      };

    case 'SORT_BY_DATE_OLDEST':
      return {
        ...state,
        sortBy: 'date(oldest)',
      };

    case 'SORT_BY_AMOUNT_HIGHEST':
      return {
        ...state,
        sortBy: 'amount(highest)',
      };

    case 'SORT_BY_AMOUNT_LOWEST':
      return {
        ...state,
        sortBy: 'amount(lowest)',
      };

    case 'SORT_BY_DESCRIPTION_A_TO_Z':
      return {
        ...state,
        sortBy: 'description(a_to_z)',
      };

    case 'SORT_BY_DESCRIPTION_Z_TO_A':
      return {
        ...state,
        sortBy: 'description(z_to_a)',
      };

    case 'SORT_BY_CATEGORY':
      return {
        ...state,
        sortBy: 'category',
      };

    case 'SET_START_DATE':
      return {
        ...state,
        startDate: action.startDate,
      };

    case 'SET_END_DATE':
      return {
        ...state,
        endDate: action.endDate,
      };

    default:
      return state;
  }
};
