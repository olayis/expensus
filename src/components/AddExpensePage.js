import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { startChangeCurrency } from '../actions/locale';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.startChangeCurrency({ currency: 'Danish Krone' });
    this.props.history.push('/');
  };
  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='add-expense-container'>
            <ExpenseForm onSubmit={this.onSubmit} />
          </div>
        </div>
      </main>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  startChangeCurrency: (currency) => dispatch(startChangeCurrency(currency)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
