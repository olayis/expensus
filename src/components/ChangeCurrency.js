import React from 'react';
import { connect } from 'react-redux';
import { startChangeLocale } from '../actions/locale';
import CurrencyForm from './CurrencyForm';

export class ChangeCurrency extends React.Component {
  onSubmit = ({ currency }) => {
    this.props.startChangeLocale({ currency });
    this.props.history.push('/');
    this.props.history.go(0);
  };

  goBack = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='component-container component-container--currency'>
            <CurrencyForm
              onSubmit={this.onSubmit}
              currency={this.props.currency}
              goBack={this.goBack}
            />
          </div>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state) => ({
  currency: state.locale ? state.locale.currency : '',
});

const mapDispatchToProps = (dispatch) => ({
  startChangeLocale: (currency) => dispatch(startChangeLocale(currency)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCurrency);
