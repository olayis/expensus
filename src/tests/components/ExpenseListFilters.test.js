import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { ExpenseListFilters } from '../../components/ExpenseListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter,
  setStartDate,
  setEndDate,
  sortByDateRecent,
  sortByDateOldest,
  sortByAmountHighest,
  sortByAmountLowest,
  sortByDescriptionAtoZ,
  sortByDescriptionZtoA,
  sortByCategory,
  wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  setStartDate = jest.fn();
  setEndDate = jest.fn();
  sortByDateRecent = jest.fn();
  sortByDateOldest = jest.fn();
  sortByAmountHighest = jest.fn();
  sortByAmountLowest = jest.fn();
  sortByDescriptionAtoZ = jest.fn();
  sortByDescriptionZtoA = jest.fn();
  sortByCategory = jest.fn();

  wrapper = shallow(
    <ExpenseListFilters
      filters={filters}
      setTextFilter={setTextFilter}
      setStartDate={setStartDate}
      setEndDate={setEndDate}
      sortByDateRecent={sortByDateRecent}
      sortByDateOldest={sortByDateOldest}
      sortByAmountHighest={sortByAmountHighest}
      sortByAmountLowest={sortByAmountLowest}
      sortByDescriptionAtoZ={sortByDescriptionAtoZ}
      sortByDescriptionZtoA={sortByDescriptionZtoA}
      sortByCategory={sortByCategory}
    />
  );
});

test('should render ExpenseListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseListFilters with alt data correctly', () => {
  wrapper.setProps({ filters: altFilters });
  expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
  const value = 'room';
  wrapper.find('input').simulate('change', {
    target: { value },
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test('should sort by recent date', () => {
  const value = 'date(recent)';
  wrapper.setProps({ filters: altFilters });
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByDateRecent).toHaveBeenCalled();
});

test('should sort by oldest date', () => {
  const value = 'date(oldest)';
  wrapper.setProps({ filters: altFilters });
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByDateOldest).toHaveBeenCalled();
});

test('should sort by highest amount', () => {
  const value = 'amount(highest)';
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByAmountHighest).toHaveBeenCalled();
});

test('should sort by lowest amount', () => {
  const value = 'amount(lowest)';
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByAmountLowest).toHaveBeenCalled();
});

test('should sort by description from A to Z', () => {
  const value = 'description(a_to_z)';
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByDescriptionAtoZ).toHaveBeenCalled();
});

test('should sort by description from Z to A', () => {
  const value = 'description(z_to_a)';
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByDescriptionZtoA).toHaveBeenCalled();
});

test('should sort by category', () => {
  const value = 'category';
  wrapper.find('select').simulate('change', {
    target: { value },
  });
  expect(sortByCategory).toHaveBeenCalled();
});

test('should handle date changes', () => {
  const value = {
    startDate: moment(0).add(3, 'months'),
    endDate: moment(0).add(20, 'years'),
  };
  wrapper.find('withStyles(DateRangePicker)').prop('onDatesChange')(value);
  expect(setStartDate).toHaveBeenLastCalledWith(value.startDate);
  expect(setEndDate).toHaveBeenLastCalledWith(value.endDate);
});

test('should handle date focus change', () => {
  const calendarFocused = 'startDate';
  wrapper.find('withStyles(DateRangePicker)').prop('onFocusChange')(
    calendarFocused
  );
  expect(wrapper.state('calendarFocused')).toBe(calendarFocused);
});
