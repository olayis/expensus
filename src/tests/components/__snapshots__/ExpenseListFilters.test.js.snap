// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListFilters correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="filter-container"
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <input
          aria-label="Search expenses"
          className="text-input text-input--search"
          name="search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <FontAwesomeIcon
          border={false}
          className="text-input--search__icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f002",
                "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
              ],
              "iconName": "search",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
      <div
        className="input-group__item dropdown"
      >
        <label
          className="text-input__label"
        >
          Sort by: 
        </label>
        <select
          aria-label="Sort expenses"
          className="dropdown--sort"
          name="sort"
          onChange={[Function]}
          value="date(recent)"
        >
          <option
            value="date(recent)"
          >
            Date (Recent)
          </option>
          <option
            value="date(oldest)"
          >
            Date (Oldest)
          </option>
          <option
            value="amount(highest)"
          >
            Amount (Highest)
          </option>
          <option
            value="amount(lowest)"
          >
            Amount (Lowest)
          </option>
          <option
            value="description(a_to_z)"
          >
            A to Z
          </option>
          <option
            value="description(z_to_a)"
          >
            Z to A
          </option>
          <option
            value="category"
          >
            Category
          </option>
        </select>
      </div>
      <div
        className="input-group__item"
      >
        <withStyles(DateRangePicker)
          anchorDirection="left"
          appendToBody={false}
          block={false}
          calendarInfoPosition="bottom"
          customArrowIcon={null}
          customCloseIcon={null}
          customInputIcon={null}
          dayPickerNavigationInlineStyles={null}
          daySize={39}
          disableScroll={false}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          endDate={null}
          endDateId="unique_end_date_id"
          endDatePlaceholderText="End Date"
          firstDayOfWeek={null}
          focusedInput={null}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepFocusOnInput={false}
          keepOpenOnDateSelect={false}
          minimumNights={1}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPosition="navPositionTop"
          navPrev={null}
          noBorder={false}
          numberOfMonths={1}
          onClose={[Function]}
          onDatesChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
            Object {
              "calendarLabel": "Calendar",
              "chooseAvailableEndDate": [Function],
              "chooseAvailableStartDate": [Function],
              "clearDates": "Clear Dates",
              "closeDatePicker": "Close",
              "dateIsSelected": [Function],
              "dateIsSelectedAsEndDate": [Function],
              "dateIsSelectedAsStartDate": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "roleDescription": "datepicker",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          readOnly={false}
          regular={false}
          renderCalendarInfo={null}
          renderDayContents={null}
          renderMonthElement={null}
          renderMonthText={null}
          renderNavNextButton={null}
          renderNavPrevButton={null}
          renderWeekHeaderElement={null}
          reopenPickerOnClearDates={false}
          required={false}
          screenReaderInputMessage=""
          showClearDates={true}
          showDefaultInputIcon={false}
          small={false}
          startDate={null}
          startDateId="unique_start_date_id"
          startDatePlaceholderText="Start Date"
          verticalHeight={null}
          verticalSpacing={22}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseListFilters with alt data correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="filter-container"
  >
    <div
      className="input-group"
    >
      <div
        className="input-group__item"
      >
        <input
          aria-label="Search expenses"
          className="text-input text-input--search"
          name="search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value="bills"
        />
        <FontAwesomeIcon
          border={false}
          className="text-input--search__icon"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f002",
                "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
              ],
              "iconName": "search",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </div>
      <div
        className="input-group__item dropdown"
      >
        <label
          className="text-input__label"
        >
          Sort by: 
        </label>
        <select
          aria-label="Sort expenses"
          className="dropdown--sort"
          name="sort"
          onChange={[Function]}
          value="amount(highest)"
        >
          <option
            value="date(recent)"
          >
            Date (Recent)
          </option>
          <option
            value="date(oldest)"
          >
            Date (Oldest)
          </option>
          <option
            value="amount(highest)"
          >
            Amount (Highest)
          </option>
          <option
            value="amount(lowest)"
          >
            Amount (Lowest)
          </option>
          <option
            value="description(a_to_z)"
          >
            A to Z
          </option>
          <option
            value="description(z_to_a)"
          >
            Z to A
          </option>
          <option
            value="category"
          >
            Category
          </option>
        </select>
      </div>
      <div
        className="input-group__item"
      >
        <withStyles(DateRangePicker)
          anchorDirection="left"
          appendToBody={false}
          block={false}
          calendarInfoPosition="bottom"
          customArrowIcon={null}
          customCloseIcon={null}
          customInputIcon={null}
          dayPickerNavigationInlineStyles={null}
          daySize={39}
          disableScroll={false}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          endDate={"1970-01-05T00:00:00.000Z"}
          endDateId="unique_end_date_id"
          endDatePlaceholderText="End Date"
          firstDayOfWeek={null}
          focusedInput={null}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepFocusOnInput={false}
          keepOpenOnDateSelect={false}
          minimumNights={1}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPosition="navPositionTop"
          navPrev={null}
          noBorder={false}
          numberOfMonths={1}
          onClose={[Function]}
          onDatesChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
            Object {
              "calendarLabel": "Calendar",
              "chooseAvailableEndDate": [Function],
              "chooseAvailableStartDate": [Function],
              "clearDates": "Clear Dates",
              "closeDatePicker": "Close",
              "dateIsSelected": [Function],
              "dateIsSelectedAsEndDate": [Function],
              "dateIsSelectedAsStartDate": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "roleDescription": "datepicker",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            }
          }
          readOnly={false}
          regular={false}
          renderCalendarInfo={null}
          renderDayContents={null}
          renderMonthElement={null}
          renderMonthText={null}
          renderNavNextButton={null}
          renderNavPrevButton={null}
          renderWeekHeaderElement={null}
          reopenPickerOnClearDates={false}
          required={false}
          screenReaderInputMessage=""
          showClearDates={true}
          showDefaultInputIcon={false}
          small={false}
          startDate={"1970-01-01T00:00:00.000Z"}
          startDateId="unique_start_date_id"
          startDatePlaceholderText="Start Date"
          verticalHeight={null}
          verticalSpacing={22}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={false}
        />
      </div>
    </div>
  </div>
</div>
`;
